clang/lib/Driver/Driver.cpp

```diff
@@ -4709,6 +4728,46 @@ class OffloadingActionBuilder final {
     return C.MakeAction<OffloadAction>(HDep, DDeps);
   }
 
+  // Update Input action to reflect FPGA device archive specifics based
+  // on archive contents.
+  bool updateInputForFPGA(Action *&A, const Arg *InputArg,
```

Comment:
Wouldn't a simple pointer type do the trick as well?
```suggestion
  bool updateInputForFPGA(Action *A, const Arg *InputArg,
```

Reply:
I don't think so, I'm passing by reference as I'm creating a new action and replacing if needed.

Reply:
Took me a while to grasp, but clear enough now, thanks! It's essentially the same mechanics as with the `addHostDependenceToDeviceActions(...)` caller.